--- a/src/video/wayland/SDL_waylandevents.c
+++ b/src/video/wayland/SDL_waylandevents.c
@@ -2776,7 +2776,7 @@
 
         size_t mime_count = 0;
         const char **text_mime_types = Wayland_GetTextMimeTypes(SDL_GetVideoDevice(), &mime_count);
-        for (size_t i = 0; i < mime_count; ++i) {
+        for (size_t i = 0; !data_device->has_mime_file && i < mime_count; ++i) {
             if (Wayland_data_offer_has_mime(data_device->drag_offer, text_mime_types[i])) {
                 data_device->has_mime_text = true;
                 data_device->mime_type = text_mime_types[i];
